client/src/main/java/org/daisy/validator/Util.java

```diff
@@ -102,6 +102,9 @@ public static boolean checkIfContainsDoctype(InputStream inputStream) throws Exc
         String fileContent = "";
         while ((line = br.readLine()) != null) {
             fileContent += line;
+            if (fileContent.matches(".*<[a-zA-Z].*")) {
```

Comment:
A corner case here would be if there's no newline after the doctype. So this would fail this check:

```html
<!DOCTYPE html><html>
<head></head><body></body></html>
```

However, I think it's good enough. It's nicer to have newlines after the doctype anyway I think.